src/plugin/vault/vault_test.go

```diff
@@ -225,51 +226,188 @@ func TestLogin(t *testing.T) {
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+}
+
+// TestLoginLongTTL tests that it's possible to get a Pachyderm token with a
+// long TTL
+func TestLoginLongTTL(t *testing.T) {
+	// Get Vault client
+	vaultClientConfig := vault.DefaultConfig()
+	vaultClientConfig.Address = vaultAddress
+	v, err := vault.NewClient(vaultClientConfig)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	v.SetToken("root")
+	vl := v.Logical()
+
+	// Get Pachyderm token with long TTL
+	secret, err := vl.Write(
+		fmt.Sprintf("/%v/login/github:bogusgithubusername", pluginName),
+		map[string]interface{}{"ttl": "768h"},
+	)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	if secret.LeaseDuration < int((700 * time.Hour).Seconds()) {
+		t.Fatalf("Expected pachyderm token with TTL=768h, but was %ds", secret.LeaseDuration)
+	}
+
+	// Make sure token is valid
+	pachToken, ok := secret.Data["user_token"].(string)
+	if !ok {
+		t.Fatalf("vault login response did not contain user token")
+	}
+	reportedPachdAddress, ok := secret.Data["pachd_address"].(string)
+	if !ok {
+		t.Fatalf("vault login response did not contain pachd address")
+	}
+	c, err := client.NewFromAddress(reportedPachdAddress)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	c.SetAuthToken(pachToken)
+	if _, err = c.AuthAPIClient.GetAdmins(c.Ctx(), &auth.GetAdminsRequest{}); err != nil {
+		t.Fatalf(err.Error())
+	}
+}
+
 func TestRenewBeforeTTLExpires(t *testing.T) {
 	ttl := 10
-	c, vaultClient, secret := loginHelper(t, fmt.Sprintf("%vs", ttl))
+	c, v, secret := loginHelper(t, fmt.Sprintf("%vs", ttl))
+	if secret.LeaseDuration < 2 {
+		t.Fatalf("expected lease to be at least 2s, but was: %d", secret.LeaseDuration)
+	} else if secret.LeaseDuration > 100 {
+		t.Fatalf("expected lease to be at most 100s, but was: %d", secret.LeaseDuration)
+	}
 
 	_, err := c.AuthAPIClient.GetAdmins(c.Ctx(), &auth.GetAdminsRequest{})
 	if err != nil {
 		t.Fatalf(err.Error())
 	}
 
-	renewer, err := vaultClient.NewRenewer(&vault.RenewerInput{
+	renewer, err := v.NewRenewer(&vault.RenewerInput{
 		Secret:    secret,
 		Increment: ttl,
 	})
 	if err != nil {
 		t.Fatalf(err.Error())
 	}
 
+	// Begin a renewer background process, and wait until it fires
 	time.Sleep(time.Duration(ttl/2) * time.Second)
 	go renewer.Renew()
 	defer renewer.Stop()
-
 	select {
 	case err := <-renewer.DoneCh():
 		if err != nil {
 			t.Fatalf(err.Error())
 		}
 	case <-renewer.RenewCh():
 	}
-	time.Sleep(time.Duration(ttl/2+1) * time.Second)
 
+	// Make sure that the vault lease was only extended by 10s
+	leaseInfo, err := v.Logical().Write("/sys/leases/lookup", map[string]interface{}{
+		"lease_id": secret.LeaseID,
+	})
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+	newDurationStr := leaseInfo.Data["ttl"].(json.Number)
+	newDuration, err := newDurationStr.Int64()
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	if newDuration < 2 {
+		t.Fatalf("expected lease to be at least 2s, but was: %d", newDuration)
+	} else if newDuration > 100 {
```

Comment:
Same thing here ... I think the max we expect is 10

Reply:
I changed this to 20 (because it's a lease that's been renewed: 10s + 10s)